src/main/lens-proxy/lens-proxy.injectable.ts

```diff
@@ -4,32 +4,24 @@
  */
 import { getInjectable } from "@ogre-tools/injectable";
 import { LensProxy } from "./lens-proxy";
-import { kubeApiUpgradeRequest } from "./proxy-functions";
 import routerInjectable from "../router/router.injectable";
 import httpProxy from "http-proxy";
-import clusterManagerInjectable from "../cluster-manager.injectable";
-import shellApiRequestInjectable from "./proxy-functions/shell-api-request/shell-api-request.injectable";
+import shellApiRequestInjectable from "./proxy-functions/shell/api-request.injectable";
 import lensProxyPortInjectable from "./lens-proxy-port.injectable";
+import getClusterForRequestInjectable from "./get-cluster-for-request.injectable";
+import kubeApiUpgradeRequestInjectable from "./proxy-functions/kube/api-upgrade-request.injectable";
 
 const lensProxyInjectable = getInjectable({
   id: "lens-proxy",
 
-  instantiate: (di) => {
-    const clusterManager = di.inject(clusterManagerInjectable);
-    const router = di.inject(routerInjectable);
-    const shellApiRequest = di.inject(shellApiRequestInjectable);
-    const proxy = httpProxy.createProxy();
-    const lensProxyPort = di.inject(lensProxyPortInjectable);
-
-    return new LensProxy({
-      router,
-      proxy,
-      kubeApiUpgradeRequest,
-      shellApiRequest,
-      getClusterForRequest: clusterManager.getClusterForRequest,
-      lensProxyPort,
-    });
-  },
+  instantiate: (di) => new LensProxy({
+    router: di.inject(routerInjectable),
+    proxy: httpProxy.createProxy(),
```

Comment:
Extract probable side-effect in injectable with `causesSideEffects`?

Reply:
I have checked and it doesn't look like it has any side effects, should I still do this?

Reply:
I think so yes. It is an external framework which currently lacks full understanding. Eg. if there's multiple places which `httpProxy.createProxy()`, will they step on each others toes? Maybe if the instances share a port or something? If there's a possibility of such, we'd do well to call it a side-effect :)

Also unit-testability would benefit from overridability of this.

Reply:
All extract to an injectable but I won't mark it as "causesSideEffects" because it doesn't seem entirely necessary yet.